### DISCUSSION TITLE
History not being properly saved

### DISCUSSION THREAD

**Discussion by filipebcs** [Eugeny/tabby#9615] [General]
My history is not saving every command and output I type as shown on the image below. Any clues on what is going on?

<img width="1030" alt="image" src="https://github.com/Eugeny/tabby/assets/11351791/e2bbc5c4-bcb8-49b4-94b5-9bc48ec13d1b">

**Comment by Eugeny** [comment]
Tabby as a terminal app has no effect on your shell's history.

**Reply by oldgithubman** [reply]
So what's the problem then? Because my history isn't being properly saved either, which is very frustrating. Also, why would you close this as "resolved," when it doesn't appear anything has been resolved?

**Reply by filipebcs** [reply]
The issue is not with Tabby itself, but rather on your .zshrc file. You should review it to check if there are no conflicts inside it that might be causing the loss of some commands in your history file. Since every user's file is different, you have to check it yourself. I organized mine and it solved the issue. Sorry for forgetting to post the answer here.

**Comment by oldgithubman** [comment]
> The issue is not with Tabby itself, but rather on your .zshrc file. You should review it to check if there are no conflicts inside it that might be causing the loss of some commands in your history file. Since every user's file is different, you have to check it yourself. I organized mine and it solved the issue. Sorry for forgetting to post the answer here.

Well, in my case it's bash, but I understand what you're saying. The problem seems to be that "ungraceful" shutdowns result in no history being saved by bash or tabby (which, AFAICT, makes no attempt to save this kind of history). So if you forget to close the window before restarting, you might lose (in my case) a month's worth of painstakingly-created commands, which, while partially my fault for multiple reasons, still really sucks and *can* be solved within tabby, if desired. And I think it should be. Is it tabby's *responsibility*? Probably not. But the end user's experience doesn't care where the responsibility lies and it would certainly make for a better experience for the user if tabby "helped" bash (or zsh, etc) more-intelligently save history. No worries about forgetting to post the answer. You're not the one who closed the issue.

Anyway, I'm evaluating "Zellij" as it seems like a replacement that handles this. I'm interested in any further information anyone has to share
